### ISSUE TITLE
Custom Color boxes not being painted

### ISSUE THREAD

**Issue by xwcg** [cyotek/Cyotek.Windows.Forms.ColorPicker#3]
In ColorGrid.cs on line 558

     if (this.ColorRegions.TryGetValue(colorCount + 1, out bounds) && e.ClipRectangle.IntersectsWith(bounds))

the +1 should be +i instead. Otherwise only the 2nd custom color will ever be painted:

     if (this.ColorRegions.TryGetValue(colorCount + i, out bounds) && e.ClipRectangle.IntersectsWith(bounds))

**Comment by cyotek** [comment]
Hello!

Funnily enough, I'm currently in the middle of adding a new demo to show how the `ColorGrid` could be used in a `ToolStrip` and I'd noticed the custom colours were not drawing properly but hadn't looked into it yet. Just checked the diff and seems I broke the line you mentioned with the last update, attention to detail fail :(

Thanks for pointing it out, saved me some time debugging the issue myself.

I'll be pushing a new update in the next day or so with some other bits and bobs, I'll make sure this fix goes in.

Thanks again!

Regards;
Richard Moss

**Comment by xwcg** [comment]
Hi,

no problem, we needed to implement your ColorPicker control into one of our projects and other than that little bug it was a perfect fit. After finding the issue, I just shot the fix up here for others who might have had problems too, but I am glad to see you're still working on it and that it's going to be fixed in the next version.

Well almost the only issue if I am honest. When you select a custom color and then change the color in a color wheel (connected via the manager) then either only the first color gets updated or a new color gets added for every color on the way when auto adding is enabled (i.e. while dragging across the color wheel trying to find the right shade). The expected behaviour would be that when you have a custom color selected, it will always change that custom color until a completely new color is picked or one clicks on another color swatch.

I managed to "kind-of" fix it, but it's pretty ugly. But if you want to, I'll open a new issue with the code that I added in order to make it work.

Best regards,
Michael Schwarz
